edit/edit.go

```diff
@@ -431,6 +436,27 @@ func sub1(ed *Editor, at addr, re *re1.Regexp, repl []rune) ([][2]int64, error)
 	return m, pend(ed, at, sliceSource(rs))
 }
 
+// nthMatch skips past the first n-1 matches of the regular expression
+func nthMatch(ed *Editor, at addr, re *re1.Regexp, n int) ([][2]int64, error) {
+	var m [][2]int64
+	var err error
+	var i int
+	for i = 0; i < n; i++ {
+		m, err = match(ed, at, re)
+		if err != nil || m == nil {
+			return nil, err
+		}
+		at.from = m[0][1]
+	}
+	if i == 0 {
```

Comment:
I think that this is more confusing and uses more code than handling it above.
Before the loop do:
`if n == 0 { n = 1 }`
Then you don't need `i` outside the loop and you can move its declaration to the for loop.